File: capitalized_name/capitalized_name.rb
require 'pry'
require 'pry-nav'

arr1 = ["ana maria", "joão silva"]
arr2 = ["pedro"]
name_composto1 = arr1.map do |name|
  partes = name.split(' ')
  first_name = partes[0].capitalize
  last_name = partes[-1].capitalize
  "#{first_name} #{last_name}"
end

name_composto2 = arr2.map do |name|
  partes = name.split('  ')
  first_name = partes[0].capitalize
  "#{first_name}"
end  

print name_composto1
print name_composto2

File: ex03_capitalized_name/capitalized_name.js
const names1 = ["Ana Maria", "João Silva"];
const names2 = ["Pedro"];

const nameComposed1 = names1.map(name => {
  const parts = name.split(' ');
  let firstName = parts[0];
  let lastName = parts[parts.length - 1];

  // Capitaliza a primeira letra do primeiro nome e sobrenome
  firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
  lastName = lastName.charAt(0).toUpperCase() + lastName.slice(1);

  return `${firstName} ${lastName}`;
});

const nameComposed2 = names2.map(name => {
  const parts = name.split(' ');
  let firstName = parts[0];

  // Capitaliza a primeira letra do nome
  firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);

  return firstName;
});

console.log(nameComposed1);
console.log(nameComposed2);
